src/categorical_algebra/Slices.jl

```diff
@@ -0,0 +1,133 @@
+module Slices
+export Slice, SliceMorphism,SliceLimit,SliceColimit, pushout_complement
+
+using AutoHashEquals
+using ...GAT
+using ..FreeDiagrams, ..Limits, ..CSets
+using ...Theories: Category
+import ...Theories: dom, codom, compose, id
+import ..Limits: limit, colimit, universal
+import ..FinSets: force, pushout_complement
+
+@auto_hash_equals struct Slice{Ob,Hom}
+  slice::Hom
+end
+
+@auto_hash_equals struct SliceMorphism{Ob, Hom}
+  dom::Slice{Ob, Hom}
+  codom::Slice{Ob, Hom}
+  f::Hom
+  function SliceMorphism{Ob, Hom}(dom, codom, f) where {Ob, Hom}
+    p1, p2 = force.([dom.slice, compose(f,codom.slice)])
```

Comment:
Since the slice construction should work for any category, and not all category `@instances` in Catlab support or know about the `force` function, this constructor won't work in general.

There is a deeper issue here, which is that in Catlab we haven't taken a consistent position on whether `==` means mathematical equality or some intermediate notion of "data structure equality," which is weaker than `===` but sometimes stricter than mathematical equality. Often it actually it means the latter. 

We are not going to solve that problem here. For now, I'll just suggest that we follow the somewhat clunky pattern used [here](https://github.com/AlgebraicJulia/Catlab.jl/blob/35cb13bec84ea3137701948c307ecf325fb56681/src/categorical_algebra/Categories.jl#L243) and elsewhere of having keyword argument `strict` that controls whether the equality condition is checked. A side benefit is that when you call the constructor in `compose` below you can set `strict=false` and avoid a superfluous equality check. (That the check is not needed is what it means for composition in a slice category to be well-defined.)